src/System.Private.CoreLib/src/System/Exception.CoreCLR.cs

```diff
@@ -420,5 +420,15 @@ internal DispatchState CaptureDispatchState()
             return new DispatchState(stackTrace, dynamicMethods,
                 _remoteStackTraceString, _ipForWatsonBuckets, _watsonBuckets);
         }
+
+        partial void SetCurrentStackTraceCore()
+        {
+            // Store the current stack trace into the "remote" stack trace, which was originally introduced
+            // to support remoting of exceptions across app-domain boundaries, and is thus concatenated into
+            // Exception.StackTrace when it's retrieved.
+            _remoteStackTraceString =
+                Environment.StackTrace + Environment.NewLine +
```

Comment:
Would prefer if this either called a `StackTrace` method with `[StackTraceHidden]` or called `new StackTrace(skipFrames: 0, fNeedFileInfo: true).ToString(...)` so it skips `System.Environment.get_StackTrace()` from the given trace?

Reply:
@danmosemsft is a little excited:
![image](https://user-images.githubusercontent.com/2642209/66133202-4413a680-e5c4-11e9-9275-3f066f5001f4.png)

Reply:
I like pretty exceptions that’s all 🙂